packages/expo-modules-autolinking/src/autolinking.ts

```diff
@@ -62,84 +62,203 @@ export async function findDefaultPathsAsync(cwd: string): Promise<string[]> {
   return paths;
 }
 
+/**
+ * Finds the real path to custom native modules directory.
+ * @returns undefined if custom modules dir not found or doesn't exist
+ */
+export async function resolveNativeModulesDirAsync(
+  nativeModulesDir: string | undefined,
+  cwd: string
+): Promise<string | undefined> {
+  // first try resolving the provided dir
+  if (nativeModulesDir) {
+    const nativeModulesDirPath = path.resolve(nativeModulesDir);
+    if (await fs.pathExists(nativeModulesDirPath)) {
+      return nativeModulesDirPath;
+    }
+  }
+
+  // if not found, try to find it relative to the package.json
+  const up = await findUp('package.json', { cwd });
+  if (!up) {
+    return undefined;
+  }
+  const resolvedPath = path.join(up, '..', nativeModulesDir || 'modules');
+  return fs.existsSync(resolvedPath) ? resolvedPath : undefined;
```

Comment:
Same here with `null`. Also, since the function is already `async`, we can probably use async version of `exists` too 😉